### DISCUSSION TITLE
How To Validate Array Values

### DISCUSSION THREAD

**Discussion by thomassmith10** [spatie/laravel-data#732] [Q&A]
Is it possible to validate array values?

For example I want to validate that every value in an array is a uuid...

```
<?php

namespace App\Dtos;

use Spatie\LaravelData\Attributes\Validation\Uuid;
use Spatie\LaravelData\Data;

class MyDTO extends Data
{
    public function __construct(
        public string $title,

        // TODO validate each value of the array by `#[Uuid]`
        public array $array_of_uuids,
    ) {
    }
}
```

Using Laravel Rules I would do it this like...

```
'array_of_uuids' => 'required|array',
'array_of_uuids.*' => 'required|uuid',
```

**Comment by rubenvanassche** [comment]
Take a look at the rules method in the docs.

**Reply by Squad-G** [reply]
Where is it ?

Say I want to validate an array of enums or an array of int ? I don't see a rule for that and the rule ArrayType doesn't support something similar to `ensure` from Laravel collection ?

I searched but can't find.

**Reply by rubenvanassche** [reply]
You can define a static rules method and then add the rules over there like you would in a regular Laravel request

**Reply by Squad-G** [reply]
Thank you.

That we knew. I should of mentioned I was looking for a way with an attribute already built in in the package.

ArrayType could be modified to do the above or a new attribute ArrayOfType

**Reply by rubenvanassche** [reply]
We'll keep the attributes as close to Laravel as possible, but you can always create your own validation attribute.

**Comment by clementbirkle** [comment]
If you want to avoid to create your own rule, you can just do it like that:

```
class MyData extends Data
{
    public function __construct(
        public array $uuids,
    ) {
    }

    public static function rules(): array
    {
        return [
            'uuids.*' => [new Required, new StringType, new Max(32)],
        ];
    }
}
```
